### ISSUE TITLE
[FEAT][CV] Add a check to spot bbox out of image bounds

### ISSUE THREAD

**Issue by TheSolY** [deepchecks/deepchecks#1556]
Integrity check.
Pass through the detection labels and make sure that 0<xmin, xmix < W or -W/2 < xmin, xmax < W/2 and 0< ymin, ymax < H or -W/2 < ymin, ymax < W/2 for all of the bounding boxes.

**Comment by noamzbr** [comment]
Actually given our architecture it's makes sense to do this not as a check but as part of validate_label. validate_label already does the "check that isn't negative part", but it can also do the rest.

**Comment by Nadav-Barak** [comment]
I agree it should be part of the validate_label, currently we have validation that coordinates are non negative.

DoD: Add a validation per image that x + width, x + height, y + width, y + height are inside image boundaries

@noamzbr note, currently we only validate format on the first image. Should we validate on all images in first batch?

**Comment by noamzbr** [comment]
would be nice to do it for a few images in the first batch, not all as the batch may contain all data.

**Comment by TheSolY** [comment]
Tasks:

- [ ] Encapsulate the code from  validate_infered_batch_predictions in a function to make sure the label bbox is inside the image boundaries - 1 SP
- [ ] Add some random sampling to for image, bbox pairs to validate on - 0.5 SP
- [ ] Add the function to validate_label - 0.5 SP
- [ ] Add a unit test
